### PR TITLE
DIP-17 changes

### PR DESCRIPTION
**Pull request by gustavoguimaraes**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,5 +105,16 @@
       "github:web3-js/scrypt-shim": false,
       "postinstall-postinstall": false
     }
-  }
+  },
+  "directories": {
```
Comment (gustavoguimaraes) [inline package.json:110]: Thanks!

File: contracts/dao/Comptroller.sol [hunk 2]
```diff
@@ -27,12 +27,13 @@ contract Comptroller is Setters {
 
     bytes32 private constant FILE = "Comptroller";
 
-    function setPrice(Decimal.D256 memory price) internal {
+    function setPrice(Decimal.D256 memory price, Decimal.D256 memory contractionPrice) internal {
```
Comment (devnulldsd) [inline contracts/dao/Comptroller.sol]: Rename to plural

File: contracts/dao/Regulator.sol [hunk 3]
```diff
@@ -70,16 +70,21 @@ contract Regulator is Comptroller {
...
+            price = Constants.getBootstrappingPrice();
         }
         if (!valid) {
-            return Decimal.one();
+            price = Decimal.one();
+        }
+
+        (Decimal.D256 memory contractionPrice, bool contractionValid) = contractionOracle().capture();
+        if (!contractionValid) {
+            contractionPrice = Decimal.one();
```
Comment (devnulldsd) [inline contracts/dao/Regulator.sol]: Would fallback to price here not 1

File: contracts/dao/State.sol [hunk 4]
```diff
@@ -126,6 +126,11 @@ contract Storage {
         uint256 totalCDSDRedeemable;
         uint256 totalCDSDRedeemed;
     }
+
+    struct State17 {
+        Decimal.D256 contractionPrice;
```
Comment (devnulldsd) [inline contracts/dao/State.sol]: Would name this CDSDPrice for clarity

**Pull request merged**